### PR TITLE
refactor(birdie): rm unused RepoConfig

### PR DESCRIPTION
**Pull request by cynful**

### CODE REVIEW

**Review by rudoi — APPROVED** [review]

File: birdie/src-tauri/src/repo/push.rs [hunk 1]
```diff
@@ -158,7 +157,7 @@ pub async fn push_handler(
         files: request.files.clone(),
         git_client: state.git(),
         repo_status: state.repo_status.clone(),
-        trunk_branch: state.repo_config.read().trunk_branch.clone(),
+        trunk_branch: "main".to_string(),
```
Comment (cynful) [inline birdie/src-tauri/src/repo/push.rs:160]: this was the thing that I was not sure was bad or not
@rudoi

Comment (cynful) [reply birdie/src-tauri/src/repo/push.rs:160]: should i write a new function for git to get the remote HEAD? and call that instead of just assigning main?

Comment (rudoi) [reply birdie/src-tauri/src/repo/push.rs:160]: IIRC we were defaulting this to main anyway so this isn't strictly worse.

Comment (cynful) [reply birdie/src-tauri/src/repo/push.rs:160]: correct it is defaulted to main

I'm going to proceed as is

**Pull request merged**